lizmap/modules/lizmap/lib/Request/WMSRequest.php

```diff
@@ -847,4 +866,292 @@ protected function gfiGmlToHtml($gmldata, $configLayer)
+                $mime = 'image/jpeg';
+                if (preg_match('#png#', $params['format'])) {
+                    $mime = 'image/png';
+                }
+
+                if ($debug) {
+                    \lizmap::logMetric('LIZMAP_PROXY_HIT_CACHE');
+                }
+
+                return array($tile, $mime, 200, true);
+            }
+        }
+
+        // ***************************
+        // No cache hit : need to ask the tile from QGIS Server
+        // ***************************
+
+        // Construction of the WMS url : base url + parameters
+        $url = $ser->wmsServerURL.'?';
+
+        // Add project path into map parameter
+        $params['map'] = $lproj->getRelativeQgisPath();
+
+        // Metatile : if needed, change the bbox
+        // Avoid metatiling when the cache is not active for the layer
+        $metatileSize = '1,1';
+        if ($configLayer and property_exists($configLayer, 'metatileSize')) {
+            if (preg_match('#^[3579],[3579]$#', $configLayer->metatileSize)) {
+                $metatileSize = $configLayer->metatileSize;
+            }
+        }
+
+        // Metatile buffer
+        $metatileBuffer = 5;
+
+        // Also checks if gd is installed
+        if ($metatileSize && $useCache && $wmsClient == 'web' &&
+            extension_loaded('gd') && function_exists('gd_info')) {
+            // Metatile Size
+            $metatileSizeExp = explode(',', $metatileSize);
+            $metatileSizeX = (int) $metatileSizeExp[0];
+            $metatileSizeY = (int) $metatileSizeExp[1];
+
+            // Get requested bbox
+            $bboxExp = explode(',', $params['bbox']);
+            $width = $bboxExp[2] - $bboxExp[0];
+            $height = $bboxExp[3] - $bboxExp[1];
+            // Calculate factors
+            $xFactor = (int) ($metatileSizeX / 2);
+            $yFactor = (int) ($metatileSizeY / 2);
+            // Calculate the new bbox
+            $xmin = $bboxExp[0] - $xFactor * $width - $metatileBuffer * $width / $params['width'];
+            $ymin = $bboxExp[1] - $yFactor * $height - $metatileBuffer * $height / $params['height'];
+            $xmax = $bboxExp[2] + $xFactor * $width + $metatileBuffer * $width / $params['width'];
+            $ymax = $bboxExp[3] + $yFactor * $height + $metatileBuffer * $height / $params['height'];
+            // Replace request bbox by metatile bbox
+            $params['bbox'] = "${xmin},${ymin},${xmax},${ymax}";
+
+            // Keep original param value
+            $originalParams = array('width' => $params['width'], 'height' => $params['height']);
+            // Replace width and height before requesting the image from qgis
+            $params['width'] = $metatileSizeX * $params['width'] + 2 * $metatileBuffer;
+            $params['height'] = $metatileSizeY * $params['height'] + 2 * $metatileBuffer;
+        }
+
+        // Build params before send the request to Qgis
+        $builtParams = http_build_query($params);
```

Comment:
Probably we could use Proxy::constructUrl() here ?

Reply:
Done in next PR.

Reply:
okay